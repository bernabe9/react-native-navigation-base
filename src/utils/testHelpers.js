import React from 'react';
import { Provider } from 'react-redux';
import Immutable from 'immutable';

import configureStore from '../store/configureStore';

const defaultStore = configureStore(Immutable.Map());

export const withStore = (WrappedComponent, store = defaultStore) => (
  <Provider store={store}>
    {WrappedComponent}
  </Provider>
);
