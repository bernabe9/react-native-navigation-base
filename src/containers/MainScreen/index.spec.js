import React from 'react';
import renderer from 'react-test-renderer';

import MainScreen from '../MainScreen';
import { withStore } from '../../utils/testHelpers';

it('renders correctly', () => {
  const tree = renderer
    .create(withStore(<MainScreen />))
    .toJSON();
  expect(tree).toMatchSnapshot();
});
