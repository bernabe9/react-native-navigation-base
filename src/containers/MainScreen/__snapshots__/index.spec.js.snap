// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F5FCFF",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    You're logged in!
  </Text>
  <View
    accessibilityComponentType="button"
    accessibilityTraits={
      Array [
        "button",
      ]
    }
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <View
      style={
        Array [
          Object {},
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "color": "#007AFF",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
          ]
        }
      >
        Logout
      </Text>
    </View>
  </View>
</View>
`;
